File: 力扣/学习计划/剑指 Offer I/020_斐波那契数列/main.go
package main

// https://leetcode-cn.com/problems/fei-bo-na-qi-shu-lie-lcof/

func fib(n int) int {
	first, second := 0, 1
	for i := 0; i < n; i++ {
		second = first + second
		first = second - first
		first %= 1000000007
	}
	return first
}

func fib2(n int) int {
	a, b := 0, 1
	for i := 0; i < n; i++ {
		b, a = a+b, b%1000000007
	}
	return a
}

File: 力扣/学习计划/剑指 Offer I/020_斐波那契数列/main.java
class Solution {
    public int fib(int n) {
        int a = 0, b = 1;
        for (int i = 0; i < n; i++) {
            int sum = (a + b) % 1000000007;
            a = b;
            b = sum;
        }
        return a;
    }
}